### PR TITLE
Omp 2743

### PR DESCRIPTION
**Pull request by baziak3**

### CODE REVIEW

File: Exception/ConsumerException.php [hunk 1]
```diff
@@ -9,19 +9,120 @@
...
 
 /**
  * Abstract Flow Exception
- * 
+ *
  * @copyright   2016 OpticsPlanet, Inc
  * @author      Eugene Boiarynov <ievgen.boiarynov@opticsplanet.com>
  */
-abstract class ConsumerException extends \Exception
+class ConsumerException extends \Exception
```
Comment (githoober) [inline Exception/ConsumerException.php:19]: Let's rename this exception to ResponseException to better reflect its purpose.

File: Exception/ConsumerException.php [hunk 2]
```diff
@@ -9,19 +9,120 @@
...
+    }
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_ACK flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
+     */
+    public static function ack($stopConsuming = false) {
```
Comment (githoober) [inline Exception/ConsumerException.php]: Please add 'create' to all fabric methods in this class.

**Review by githoober — CHANGES_REQUESTED** [review]
Left comments about a few issues developer and I agreed on fixing.

File: Exception/ConsumerException.php [hunk 3]
```diff
@@ -9,19 +9,120 @@
  */
 
 namespace Ecentria\Libraries\EcentriaAPIEventsBundle\Exception;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
 
 /**
  * Abstract Flow Exception
```
Comment (githoober) [inline Exception/ConsumerException.php]: Please update the exception description.

File: Exception/ConsumerException.php [hunk 4]
```diff
@@ -9,19 +9,120 @@
...
- * 
+ *
  * @copyright   2016 OpticsPlanet, Inc
  * @author      Eugene Boiarynov <ievgen.boiarynov@opticsplanet.com>
  */
-abstract class ConsumerException extends \Exception
+class ConsumerException extends \Exception
 {
     /**
+     * Message status flag
```
Comment (githoober) [inline Exception/ConsumerException.php]: Maybe response status flag? List all values it can recieve

**Review by githoober — CHANGES_REQUESTED** [review]

File: Exception/ResponseException.php [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+ * This file is part of the ecentria group, inc. software.
+ *
+ * (c) 2016, ecentria group, inc.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Ecentria\Libraries\EcentriaAPIEventsBundle\Exception;
+use OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface;
```
Comment (githoober) [inline Exception/ResponseException.php:12]: Add a blank space after the namespace declaration.

File: Exception/ResponseException.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+     * @var bool
+     */
+    private $stopConsuming;
+
+    /**
+     * Constructor
+     *
+     * @param string $flag          One of the message flag defined
+     *                              inside \OldSound\RabbitMqBundle\RabbitMq\ConsumerInterface
+     * @param bool   $stopConsuming Whether it is necessary to stop the consumer or not
```
Comment (githoober) [inline Exception/ResponseException.php:47]: Missing @throws tag in function comment

File: Exception/ResponseException.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_ACK flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
+     */
+    public static function createAck($stopConsuming = false) {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_REJECT flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
+     */
+    public static function createReject($stopConsuming = false) {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_REJECT_REQUEUE flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
+     */
+    public static function createdRejectAndRequeue($stopConsuming = false) {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_SINGLE_NACK_REQUEUE flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
+     */
+    public static function createNackAndRequeue($stopConsuming = false) {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+        return new self(ConsumerInterface::MSG_SINGLE_NACK_REQUEUE, $stopConsuming);
+    }
+
+    /**
+     * Returns a flag value
+     *
+     * @return int
+     */
+    public function getFlag() {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+        return $this->flag;
+    }
+
+    /**
+     * Returns true if it is necessary to stop the consumer, false otherwise
+     *
+     * @return bool
+     */
+    public function stopConsuming() {
```
Comment (githoober) [inline Exception/ResponseException.php]: Opening brace should be on a new line

File: Exception/ResponseException.php [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+        return new self(ConsumerInterface::MSG_REJECT, $stopConsuming);
+    }
+
+
+    /**
+     * Creates exception with ConsumerInterface::MSG_REJECT_REQUEUE flag
+     *
+     * @param bool $stopConsuming Whether it is necessary to stop the consumer or not
+     *
+     * @return ConsumerException
```
Comment (githoober) [inline Exception/ResponseException.php]: ResponseException  here and in all other fabric  methods.

**Review by githoober — CHANGES_REQUESTED** [review]
Left comments about formatting based on OPPSR-2

File: Exception/ConsumerException.php (hunk 1 above)
Comment (baziak3) [reply Exception/ConsumerException.php:19]: Addressed

File: Exception/ResponseException.php (hunk 6 above)
Comment (baziak3) [reply Exception/ResponseException.php:47]: Addressed

File: Exception/ResponseException.php (hunk 12 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 11 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 10 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 9 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 13 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 8 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 7 above)
Comment (baziak3) [reply Exception/ResponseException.php]: Addressed

File: Exception/ResponseException.php (hunk 5 above)
Comment (baziak3) [reply Exception/ResponseException.php:12]: Addressed

File: Exception/ConsumerException.php (hunk 4 above)
Comment (baziak3) [reply Exception/ConsumerException.php]: Addressed

File: Exception/ConsumerException.php (hunk 3 above)
Comment (baziak3) [reply Exception/ConsumerException.php]: Addressed

**Review by githoober — APPROVED** [review]

**Pull request merged**